nucleus/core/kernel/src/main/java/org/glassfish/runnablejar/commandline/Option.java

```diff
@@ -0,0 +1,264 @@
+        }
+    }
+
+    public abstract void handle(String value, Arguments arguments);
+
+    public String getMainName() {
+        return mainName;
+    }
+
+    public Set<String> getAliases() {
+        return aliases;
+    }
+
+    public String getHelpText() {
+        return helpText;
+    }
+
+    public String getUsage() {
+        return usage;
+    }
+
+    protected static boolean mainNameMatches(Option option, String upperCaseKey) {
+        return option.mainName.toUpperCase().equals(upperCaseKey);
+    }
+
+    protected static boolean aliasesMatch(Option option, String upperCaseKey) {
+        return option.aliases.stream().filter(alias -> alias.toUpperCase().equals(upperCaseKey)).findAny().isPresent();
+    }
+
+    protected void setPort(String listener, String value, Arguments arguments) throws NumberFormatException {
+        setPort(listener, Integer.parseInt(value), arguments);
+    }
+
+    protected void setPort(String listener, int value, Arguments arguments) throws NumberFormatException {
+        arguments.glassFishProperties.setPort(listener, value);
+    }
+
+    protected void loadPropertiesFromFile(String fileName, Arguments arguments) {
+        try {
+            final Properties properties = new Properties();
+            properties.load(Files.newBufferedReader(Paths.get(fileName)));
+            properties.forEach((k, v) -> {
+                try {
+                    arguments.setOption((String) k, (String) v);
+                } catch (UnknownPropertyException e) {
+                    logger.log(Level.WARNING, e, () -> "Invalid property '" + e.getKey() + "' in file " + fileName);
+                }
+            });
+        } catch (IOException e) {
+            logger.log(Level.WARNING, e, () -> "Could not read properties from file " + fileName + " - " + e.getMessage());
+        }
+    }
+
+    protected void loadApplicationsFromDirectory(String directoryName, Arguments arguments) {
+        final File directory = new File(directoryName);
+        if (directory.isDirectory()) {
+            for (File appFile : directory.listFiles()) {
+                arguments.deployables.add(appFile.getAbsolutePath());
+            }
+        } else {
+            logger.log(WARNING, () -> "The path specified with the " + this.getUsage() + " option is not a directory: " + directoryName);
+        }
+    }
+
+    protected void setLogLevel(String logLevel, Arguments arguments) {
+        final Logger rootLogger = Logger.getLogger("");
+        final Level level;
+        try {
+            level = Level.parse(logLevel);
+        } catch (IllegalArgumentException | NullPointerException e) {
+            throw new IllegalArgumentException("The value of the " + this.getMainName() + " option is not valid: " + logLevel);
+        }
+        for (Handler handler : rootLogger.getHandlers()) {
+            handler.setLevel(level);
+        }
+        LogManager.getLogManager().getLoggerNames().asIterator()
```

Comment:
Problem with this is that it returns just already initialized loggers. It would be better to retrieve current configuration from the logmanager, change it, and then call logManager.readConfiguration. Or just provide our own defaults generated using this parameter, set the root logger+handler level and just call logger.readConfiguration asap, even before main (in static init). It would be done just if user would not explicitly set the `-Djava.util.logging.config.file` or `-Djava.util.logging.config.class.`

In the video it seems that currently it uses JDK defaults.